suit_generator/cmd_cache_create.py

```diff
@@ -6,116 +6,297 @@
 
+    def add_cache_slot(self, uri: str, data: bytes):
+        """
+        Add a cache slot to the cache from the given URI and data.
+
+        This function creates a cache slot from the given URI and data, and pads the data to align with the specified
+        erase block size (eb_size). The first slot in the cache is created with indefinite length CBOR map.
+
+        :param uri: The URI associated with the data.
+        :type uri: str
+        :param data: The data to be included in the cache slot.
+        :type data: bytes
+        """
         slot_data = bytes()
-        if first_slot:
+        if self.first_slot:
             # Open the cache - it is an indefinite length CBOR map (0xBF)
             slot_data = bytes([0xBF])
-            first_slot = False
+        self.first_slot = False
 
         # uri as key
         slot_data += cbor2.dumps(uri)
 
         # Size must be encoded in 4 bytes, thus cannot use cbor2.dumps
         slot_data += bytes([0x5A]) + len(data).to_bytes(4, byteorder="big") + data
         # Add padding for single slot
-        slot_data = add_padding(slot_data, eb_size)
+        slot_data = self.add_padding(slot_data)
+
+        self.cache_data += slot_data
+
+    def close_and_save_cache(self, output_file: str):
+        """
+        Close the cache and save it to the specified output file.
+
+        This function closes the cache by adding the end-of-map byte (0xFF) and saves the cache to the specified output
+        file.
+
+        :param output_file: Path to the output raw SUIT DFU cache file.
+        :type output_file: str
+        """
+        self.cache_data += bytes([0xFF])
+        with open(output_file, "wb") as f:
+            f.write(self.cache_data)
+
+    def merge_single_cache_file(self, cache_input_file: str):
+        """
+        Merge the contents of the provided single cache file into the current cache.
+        :param cache_input_file: Path to the input raw SUIT DFU cache file.
+        """
+
+        with open(cache_input_file, "rb") as f:
+            data = f.read()
+
+        cache_dict = cbor2.loads(data)
+
+        for k in cache_dict.keys():
+            if  k == b'':
+                continue # Empty key means padding - skip
+            self.add_cache_slot(k, cache_dict[k])
+
+
+class CacheFromPayloads:
+    def fill_cache_from_payloads(cache: CachePartition, input: list[str]) -> None:
+        """
+        Process list of input binaries, each associated with a URI, and fill the SUIT DFU cache with the data
+
+        :param cache: CachePartition object to fill with the data
+        :param input: List of input binaries with corresponding URIs, passed in the format <URI>,<INPUT_FILE_PATH>
+        """
+
+        for single_input in input:
+            args = single_input.split(",")
+            if len(args) < 2:
+                raise ValueError("Invalid number of input arguments: " + single_input)
+            uri, input_file = args
```

Comment:
Please, consider validation of uri's duplication.

Reply:
Added such validation